gems/pending/VMwareWebService/MiqPbmInventory.rb

```diff
@@ -50,20 +50,21 @@ def pbmQueryAssociatedEntity(profile_ids)
     assoc_entities
   end
 
-  def pbmQueryMatchingHub(profile_id)
-    hubs = []
+  def pbmQueryMatchingHub(profile_ids)
+    hubs = {}
     return hubs if @pbm_svc.nil?
 
     begin
-      # If a string was passed in create a PbmProfileId object
-      profile_id = RbVmomi::PBM::PbmProfileId(:uniqueId => profile_id) if profile_id.kind_of?(String)
+      profile_ids.each do |profile_id|
+        # If a string was passed in create a PbmProfileId object
+        profile_id = RbVmomi::PBM::PbmProfileId(:uniqueId => profile_id) if profile_id.kind_of?(String)
```

Comment:
For the storage policy stuff yes, SPBM is a different endpoint than VIM so we would have had to generate a new handsoap service or use this.
The PBM endpoint doesn't do its own authentication, instead it takes an authenticated vim session (which we get from the broker)